evpn : add autort as option

aderumier opened the pull request:
Allow to defined a custom AS for autort

Usefull for evpn with ebgp, where we can't auto-derivated from main AS

Signed-off-by: Alexandre Derumier <aderumier@odiso.com>

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/0f8318389a6c97bbaae80d4108b4b6d5/raw/4ec0f466f7827d83c360ade30d013c3e3f4ef14a/cr_7535_1605590110.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgp_evpn.c b/bgpd/bgp_evpn.c
index af16070d1..73acc5d7b 100644
--- a/bgpd/bgp_evpn.c
+++ b/bgpd/bgp_evpn.c
@@ -513,9 +513,9 @@ static void form_auto_rt(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-	    encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
-	    encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
 
 	ecomadd = ecommunity_new();
@@ -4243,9 +4243,9 @@ void evpn_rt_delete_auto(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-	    encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
-	    encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
 
 	ecom_auto = ecommunity_new();
diff --git a/bgpd/bgp_evpn_vty.c b/bgpd/bgp_evpn_vty.c
index 115a7b3d3..2e33b3c5e 100644
--- a/bgpd/bgp_evpn_vty.c
+++ b/bgpd/bgp_evpn_vty.c
@@ -3226,12 +3226,11 @@ DEFUN (no_bgp_evpn_advertise_all_vni,
 	return CMD_SUCCESS;
 }
 
-DEFPY (bgp_evpn_advertise_autort,
-       bgp_evpn_advertise_autort_cmd,
-       "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
+      "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 
@@ -3249,13 +3248,12 @@ DEFPY (bgp_evpn_advertise_autort,
 	return CMD_SUCCESS;
 }
 
-DEFPY (no_bgp_evpn_advertise_autort,
-       no_bgp_evpn_advertise_autort_cmd,
-       "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       NO_STR
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(no_bgp_evpn_advertise_autort, no_bgp_evpn_advertise_autort_cmd,
+      "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      NO_STR
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

ton31337 reviewed (changes requested):
1. Commit message should be `bgpd: ...`;
2. Topotest is missing for this case;
3. New CLI commands MUST be documented.

aderumier commented:
> 1. Commit message should be `bgpd: ...`;
> 2. Topotest is missing for this case;
> 3. New CLI commands MUST be documented.

thanks.  I'll rework the commit message && doc. I'll look for topotest. (This is my first patch

aderumier commented:
@ton31337 @donaldsharp 

I'm currently looking at current doc, but they are almost nothing for evpn configuration.
It's missing doc for basic things like "advertise-all-vni", or "route-target", or even the "autort rfc8365-compatible" command that I'm currently extending with new "autort as .." option.

So I don't known where to begin... (I don't have expertise to write the full evpn doc ;)

ton31337 commented:
It would be enough to add your command, no need to add other commands related to EVPN.

pjdruddy commented:
Agree with @ton31337 here. Add this command to the BGP user docs. Will plan to get this fully documented along with the EVPN datamodel work.

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/e927b3de125af0ecd5fe08096fe88dfa/raw/4ec0f466f7827d83c360ade30d013c3e3f4ef14a/cr_7535_1605907878.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgp_evpn.c b/bgpd/bgp_evpn.c
index af16070d1..73acc5d7b 100644
--- a/bgpd/bgp_evpn.c
+++ b/bgpd/bgp_evpn.c
@@ -513,9 +513,9 @@ static void form_auto_rt(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-	    encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
-	    encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
 
 	ecomadd = ecommunity_new();
@@ -4243,9 +4243,9 @@ void evpn_rt_delete_auto(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-	    encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
-	    encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
 
 	ecom_auto = ecommunity_new();
diff --git a/bgpd/bgp_evpn_vty.c b/bgpd/bgp_evpn_vty.c
index 115a7b3d3..2e33b3c5e 100644
--- a/bgpd/bgp_evpn_vty.c
+++ b/bgpd/bgp_evpn_vty.c
@@ -3226,12 +3226,11 @@ DEFUN (no_bgp_evpn_advertise_all_vni,
 	return CMD_SUCCESS;
 }
 
-DEFPY (bgp_evpn_advertise_autort,
-       bgp_evpn_advertise_autort_cmd,
-       "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
+      "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 
@@ -3249,13 +3248,12 @@ DEFPY (bgp_evpn_advertise_autort,
 	return CMD_SUCCESS;
 }
 
-DEFPY (no_bgp_evpn_advertise_autort,
-       no_bgp_evpn_advertise_autort_cmd,
-       "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       NO_STR
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(no_bgp_evpn_advertise_autort, no_bgp_evpn_advertise_autort_cmd,
+      "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      NO_STR
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (dismissed):
Thanks for your contribution to FRR!

* One of your commits has an improperly formatted commit message
* One of your commits has a missing or badly formatted `Signed-off-by` line; we can't accept your contribution until all of your commits have one

<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/9877b85e078b3cc650d685e04e36c8b3/raw/4ec0f466f7827d83c360ade30d013c3e3f4ef14a/cr_7535_1605909164.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgp_evpn.c b/bgpd/bgp_evpn.c
index af16070d1..73acc5d7b 100644
--- a/bgpd/bgp_evpn.c
+++ b/bgpd/bgp_evpn.c
@@ -513,9 +513,9 @@ static void form_auto_rt(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-	    encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
-	    encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
 
 	ecomadd = ecommunity_new();
@@ -4243,9 +4243,9 @@ void evpn_rt_delete_auto(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-	    encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
-	    encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
 
 	ecom_auto = ecommunity_new();
diff --git a/bgpd/bgp_evpn_vty.c b/bgpd/bgp_evpn_vty.c
index 115a7b3d3..2e33b3c5e 100644
--- a/bgpd/bgp_evpn_vty.c
+++ b/bgpd/bgp_evpn_vty.c
@@ -3226,12 +3226,11 @@ DEFUN (no_bgp_evpn_advertise_all_vni,
 	return CMD_SUCCESS;
 }
 
-DEFPY (bgp_evpn_advertise_autort,
-       bgp_evpn_advertise_autort_cmd,
-       "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
+      "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 
@@ -3249,13 +3248,12 @@ DEFPY (bgp_evpn_advertise_autort,
 	return CMD_SUCCESS;
 }
 
-DEFPY (no_bgp_evpn_advertise_autort,
-       no_bgp_evpn_advertise_autort_cmd,
-       "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       NO_STR
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(no_bgp_evpn_advertise_autort, no_bgp_evpn_advertise_autort_cmd,
+      "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      NO_STR
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (changes requested):
Thanks for your contribution to FRR!

* One of your commits has an improperly formatted commit message
* One of your commits has a missing or badly formatted `Signed-off-by` line; we can't accept your contribution until all of your commits have one

If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (changes requested):
Thanks for your contribution to FRR!

* One of your commits has an improperly formatted commit message
* One of your commits has a missing or badly formatted `Signed-off-by` line; we can't accept your contribution until all of your commits have one

If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (changes requested):
Thanks for your contribution to FRR!

* One of your commits has an improperly formatted commit message
* One of your commits has a missing or badly formatted `Signed-off-by` line; we can't accept your contribution until all of your commits have one

<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/01fae0c759adaf8f643de3cb6231a69c/raw/25d9840b21a97415fa30688911cb5bd756a95a4a/cr_7535_1605910287.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgp_evpn.c b/bgpd/bgp_evpn.c
index 99e04442f..52ae57037 100644
--- a/bgpd/bgp_evpn.c
+++ b/bgpd/bgp_evpn.c
@@ -513,9 +513,9 @@ static void form_auto_rt(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-	    encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
-	    encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
 
 	ecomadd = ecommunity_new();
@@ -4243,9 +4243,9 @@ void evpn_rt_delete_auto(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-	    encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
-	    encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
 
 	ecom_auto = ecommunity_new();
diff --git a/bgpd/bgp_evpn_vty.c b/bgpd/bgp_evpn_vty.c
index 115a7b3d3..2e33b3c5e 100644
--- a/bgpd/bgp_evpn_vty.c
+++ b/bgpd/bgp_evpn_vty.c
@@ -3226,12 +3226,11 @@ DEFUN (no_bgp_evpn_advertise_all_vni,
 	return CMD_SUCCESS;
 }
 
-DEFPY (bgp_evpn_advertise_autort,
-       bgp_evpn_advertise_autort_cmd,
-       "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
+      "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 
@@ -3249,13 +3248,12 @@ DEFPY (bgp_evpn_advertise_autort,
 	return CMD_SUCCESS;
 }
 
-DEFPY (no_bgp_evpn_advertise_autort,
-       no_bgp_evpn_advertise_autort_cmd,
-       "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       NO_STR
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(no_bgp_evpn_advertise_autort, no_bgp_evpn_advertise_autort_cmd,
+      "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      NO_STR
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (dismissed):
Thanks for your contribution to FRR!

* One of your commits has an improperly formatted commit message
* One of your commits has a missing or badly formatted `Signed-off-by` line; we can't accept your contribution until all of your commits have one

<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/dc1edd5e88a00e8dbcf5e16b7d245bed/raw/25d9840b21a97415fa30688911cb5bd756a95a4a/cr_7535_1605910492.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgp_evpn.c b/bgpd/bgp_evpn.c
index 99e04442f..52ae57037 100644
--- a/bgpd/bgp_evpn.c
+++ b/bgpd/bgp_evpn.c
@@ -513,9 +513,9 @@ static void form_auto_rt(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-	    encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
-	    encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
 
 	ecomadd = ecommunity_new();
@@ -4243,9 +4243,9 @@ void evpn_rt_delete_auto(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-	    encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
-	    encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
 
 	ecom_auto = ecommunity_new();
diff --git a/bgpd/bgp_evpn_vty.c b/bgpd/bgp_evpn_vty.c
index 115a7b3d3..2e33b3c5e 100644
--- a/bgpd/bgp_evpn_vty.c
+++ b/bgpd/bgp_evpn_vty.c
@@ -3226,12 +3226,11 @@ DEFUN (no_bgp_evpn_advertise_all_vni,
 	return CMD_SUCCESS;
 }
 
-DEFPY (bgp_evpn_advertise_autort,
-       bgp_evpn_advertise_autort_cmd,
-       "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
+      "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 
@@ -3249,13 +3248,12 @@ DEFPY (bgp_evpn_advertise_autort,
 	return CMD_SUCCESS;
 }
 
-DEFPY (no_bgp_evpn_advertise_autort,
-       no_bgp_evpn_advertise_autort_cmd,
-       "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       NO_STR
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(no_bgp_evpn_advertise_autort, no_bgp_evpn_advertise_autort_cmd,
+      "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      NO_STR
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (changes requested):
Thanks for your contribution to FRR!

* One of your commits has an improperly formatted commit message

<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/9cf33d30a2a9ad074ca29137f5de3d19/raw/738e27432e1be435d9f6a7d235ae9c4c92a6e9ef/cr_7535_1605910944.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgp_evpn.c b/bgpd/bgp_evpn.c
index ccd121d7b..52ae57037 100644
--- a/bgpd/bgp_evpn.c
+++ b/bgpd/bgp_evpn.c
@@ -4243,7 +4243,7 @@ void evpn_rt_delete_auto(struct bgp *bgp, vni_t vni, struct list *rtl)
 	if (bgp->advertise_autort_rfc8365)
 		vni |= EVPN_AUTORT_VXLAN;
 	if (bgp->autort_as) {
-		 encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
+		encode_route_target_as((bgp->autort_as & 0xFFFF), vni, &eval);
 	} else {
 		encode_route_target_as((bgp->as & 0xFFFF), vni, &eval);
 	}
diff --git a/bgpd/bgp_evpn_vty.c b/bgpd/bgp_evpn_vty.c
index 115a7b3d3..2e33b3c5e 100644
--- a/bgpd/bgp_evpn_vty.c
+++ b/bgpd/bgp_evpn_vty.c
@@ -3226,12 +3226,11 @@ DEFUN (no_bgp_evpn_advertise_all_vni,
 	return CMD_SUCCESS;
 }
 
-DEFPY (bgp_evpn_advertise_autort,
-       bgp_evpn_advertise_autort_cmd,
-       "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
+      "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 
@@ -3249,13 +3248,12 @@ DEFPY (bgp_evpn_advertise_autort,
 	return CMD_SUCCESS;
 }
 
-DEFPY (no_bgp_evpn_advertise_autort,
-       no_bgp_evpn_advertise_autort_cmd,
-       "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       NO_STR
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(no_bgp_evpn_advertise_autort, no_bgp_evpn_advertise_autort_cmd,
+      "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      NO_STR
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (changes requested):
Thanks for your contribution to FRR!

* One of your commits has an improperly formatted commit message

<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/08375632c708602f44ea38b09e24b9b1/raw/73838270947018d7b2f6eb25591fe07e4600a111/cr_7535_1605911111.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgp_evpn_vty.c b/bgpd/bgp_evpn_vty.c
index 07629b63c..2e33b3c5e 100644
--- a/bgpd/bgp_evpn_vty.c
+++ b/bgpd/bgp_evpn_vty.c
@@ -3226,11 +3226,11 @@ DEFUN (no_bgp_evpn_advertise_all_vni,
 	return CMD_SUCCESS;
 }
 
-DEFPY (bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
-       "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
+      "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 
@@ -3248,12 +3248,12 @@ DEFPY (bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
 	return CMD_SUCCESS;
 }
 
-DEFPY (no_bgp_evpn_advertise_autort, no_bgp_evpn_advertise_autort_cmd,
-       "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       NO_STR
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+DEFPY(no_bgp_evpn_advertise_autort, no_bgp_evpn_advertise_autort_cmd,
+      "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      NO_STR
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (changes requested):
Thanks for your contribution to FRR!

* One of your commits has an improperly formatted commit message

<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/4bf48e8c17f81c55c59750cad5243059/raw/d799c1fdc6b4125d80847f46dacc4aec024bd628/cr_7535_1605911259.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgp_evpn_vty.c b/bgpd/bgp_evpn_vty.c
index 7e7483b30..2e33b3c5e 100644
--- a/bgpd/bgp_evpn_vty.c
+++ b/bgpd/bgp_evpn_vty.c
@@ -3227,10 +3227,10 @@ DEFUN (no_bgp_evpn_advertise_all_vni,
 }
 
 DEFPY(bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
-       "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+      "autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 
@@ -3249,11 +3249,11 @@ DEFPY(bgp_evpn_advertise_autort, bgp_evpn_advertise_autort_cmd,
 }
 
 DEFPY(no_bgp_evpn_advertise_autort, no_bgp_evpn_advertise_autort_cmd,
-       "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
-       NO_STR
-       "Auto-derivation of RT\n"
-       "Auto-derivation of RT using RFC8365\n"
-       "Auto-derivation AS\n")
+      "no autort [rfc8365-compatible]$rfc8365 [as (1-65536)$as]",
+      NO_STR
+      "Auto-derivation of RT\n"
+      "Auto-derivation of RT using RFC8365\n"
+      "Auto-derivation AS\n")
 {
 	struct bgp *bgp = VTY_GET_CONTEXT(bgp);
 

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (changes requested):
Thanks for your contribution to FRR!

* One of your commits has an improperly formatted commit message

If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (changes requested):
Thanks for your contribution to FRR!

* One of your commits has a missing or badly formatted `Signed-off-by` line; we can't accept your contribution until all of your commits have one
* One of your commits does not have a blank line between the summary and body; this will break `git log --oneline`

If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

aderumier commented:
@ton31337 

Hi, I have added the docs && topotest for autort command. (I have also added test/doc for the previous "autort rfc8365-compatible")

and rewrite the commit msg with "bgp: ..."

pguibert6WIND commented:
Hi @aderumier , any updates on this ?

dshleg commented:
Hi all, this actual for me. UP
